File: 1338 - Reduce Array Size to The Half/solution.py
from collections import Counter
class Solution:
    def minSetSize(self, A: List[int]) -> int:
        res, n, A = 0, len(A)//2, sorted(Counter(A).values(), reverse=True)
        while n > 0:
            n -= A[res]
            res += 1
        return res

File: 1338 - Reduce Array Size to The Half/solution.cpp
class Solution {
public:
    int minSetSize(vector<int>& A) {
        int res = 0, n = A.size(), h = n/2;
        vector<int> cnt(n+1);
        unordered_map<int, int> m;
        for(auto i : A) m[i]++;
        for(auto [i, v] : m) cnt[v]++;
        while(h > 0){
            while(cnt[n] == 0) n--;
            res++, cnt[n]--, h -= n;
        }
        return res;
    }
};